core/utils/util.c

```diff
@@ -58,7 +58,7 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  * The ID of this federate. For a non-federated execution, this will be -1.
  * For a federated execution, it will be assigned in the generated code.
  */
-int _lf_my_fed_id = -1;
+int32_t _lf_my_fed_id = -1;
```

Comment:
Why not make this a 16 bit unsigned int?

```suggestion
uint16_t _lf_my_fed_id = -1;
```

Reply:
`uint16_t _lf_my_fed_id = -1;` will not work, since it is an uint16_t.

So, it seems to initialize the `_lf_my_fed_id` as -1, to ensure when it is used, it is a proper integer larger than zero. However, I don't exactly see if there are any fail cases that the `_lf_my_fed_id` will be left as -1.

Reply:
I think it would be safe to remove the -1 initialization of `_lf_my_fed_id`. It seems this variable is set in generated code in the function that initializes the federate, so it should never go uninitialized.